src/renderer/components/NewConnectionForm.tsx

```diff
@@ -143,12 +143,13 @@ const NewConnectionForm = () => {
 
     try {
       // creates the server connection in sqlite and connects to it
-      await window.connections.ipcRenderer.create(connection);
+      const db = await window.connections.ipcRenderer.create(connection);
 
       // verify if server connection was established
       const verified = await window.connections.ipcRenderer.verify();
       if (!verified) {
         setAlert(true);
+        await window.connections.ipcRenderer.delete(db.id);
```

Comment:
Yeah you're right this should also be done in the backend, but it is sort of hard to do with the way we've set it up right now, so we will change this when we refactor the redux state